src/wallet/rpcwallet.cpp

```diff
@@ -2681,76 +2681,53 @@ static UniValue upgradetohd(const JSONRPCRequest& request)
         },
     }.Check(request);
 
-    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
-    if (!wallet) return NullUniValue;
-    CWallet* const pwallet = wallet.get();
-    LegacyScriptPubKeyMan* spk_man = pwallet->GetLegacyScriptPubKeyMan();
-    if (!spk_man) {
-        throw JSONRPCError(RPC_WALLET_ERROR, "This type of wallet does not support this command");
-    }
+    std::shared_ptr<CWallet> const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!pwallet) return NullUniValue;
 
     bool generate_mnemonic = request.params[0].isNull() || request.params[0].get_str().empty();
-
-    {
-        LOCK(pwallet->cs_wallet);
-
-        // Do not do anything to HD wallets
-        if (pwallet->IsHDEnabled()) {
-            throw JSONRPCError(RPC_WALLET_ERROR, "Cannot upgrade a wallet to HD if it is already upgraded to HD.");
-        }
-
-        if (!pwallet->SetMaxVersion(FEATURE_HD)) {
-            throw JSONRPCError(RPC_WALLET_ERROR, "Cannot downgrade wallet");
+    SecureString secureWalletPassphrase;
+    secureWalletPassphrase.reserve(100);
```

Comment:
The GUI [reserves](https://github.com/bitcoin/bitcoin/blame/411ba32af21a56efa0a570b6aa8bf8f035410230/src/qt/askpassphrasedialog.cpp#L89-L91) [`MAX_PASSPHRASE_SIZE`](https://github.com/dashpay/dash/blob/ce67e07294c90bcd9a0ffb038bf8aaad0670775e/src/qt/guiconstants.h#L15) for the wallet. 

It's unlikely someone will actually come around with a password >100 characters but since we allow longer passwords in the GUI, we should have RPC behave in the same way.

Reply:
This call `reserve` has been introduced before introducing SecureString (which use custom allocator).
That call is needed to be sure that string is not accidentally re-allocated and it would be impossible to wipe a data by rewriting bytes.

This call is just for legacy and completely useless now. I'd better to consider removing `reserve` completely rather than keep it here.

btw, that's not new code, I just moved it a bit; I will consider removing `reserve` at all, not planning to refactor it in this PR